cli/internal/codegen/golang.go

```diff
@@ -17,16 +17,33 @@ import (
 	schema "encr.dev/proto/encore/parser/schema/v1"
 )
 
+// tsGenVersion allows us to introduce breaking changes in the generated code but behind a switch
+// meaning that people with client code reliant on the old behaviour can continue to generate the
+// old code.
+type goGenVersion int
+
+const (
+	// GoInitial is the originally released typescript generator
+	GoInitial goGenVersion = iota
+
+	// GoExperimental can be used to lock experimental or uncompleted features in the generated code
+	// It should always be the last item in the enum
+	GoExperimental
+)
+
+const goGenLatestVersion = GoExperimental - 1
+
 type golang struct {
-	md  *meta.Data
-	enc *gocodegen.MarshallingCodeGenerator
+	md               *meta.Data
+	enc              *gocodegen.MarshallingCodeGenerator
+	generatorVersion goGenVersion
 }
 
 func (g *golang) Generate(buf *bytes.Buffer, appSlug string, md *meta.Data) (err error) {
 	g.md = md
 	// note we use unicode characters to write "encore" to ensure we don't conflict with any existing names in the
 	// package the generated client is placed.
-	g.enc = gocodegen.NewMarshallingCodeGenerator("ℯ𝓃𝑐ℴ𝑟ℯMarshaller")
+	g.enc = gocodegen.NewMarshallingCodeGenerator("ℯ𝓃𝑐ℴ𝑟ℯMarshaller", true)
```

Comment:
How is the unicode name used? If it's in the generated code I think we should maybe go for something a bit more standard, even though it's technically allowed by the Go spec it might cause people to trip up. Maybe "encoreClientMarshaller" or something?

Reply:
@ekerfelt also commented on this in the original PR. The reasoning was simply to avoid naming complications by putting an encore prefix on it.

Happy to change to just `marshaller` or `serde` if you'd prefer (and ignore the name conflict (as I think it's a minor risk)

Reply:
Yeah I think it might trip people up; maybe something longer and more specific would minimize the risk still. I guess we probably expect people to put it in a separate package regardless.